src/librustc/mir/interpret/error.rs

```diff
@@ -516,6 +513,35 @@ impl fmt::Debug for ResourceExhaustionInfo {
     }
 }
 
+/// A trait for machine-specific errors (or other "machine stop" conditions).
+pub trait MachineStopType: Any + fmt::Debug + Send {}
+impl MachineStopType for String {}
+
+// Copy-pasted from `any.rs`; there does not seem to be a way to re-use that.
```

Comment:
Ugh... if we had https://github.com/rust-lang/rust/pull/60900 this would not be necessary

Reply:
Here's a workaround trick that doesn't require you to reimplement `Any`: https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=dfc0e0f6769770dd4984ec6b30a8dae7

Reply:
Nice! I tried something like that but was not creative enough.